next/docs/wasp/docs/guide/core_concepts/state_manager.md

```diff
@@ -38,78 +38,86 @@ contains the same commitment as state index `n`.
-be consider as being the same.
+5. the mentioned block's hash in `20` bytes,
+6. number `0` in 1 byte unsigned integer format representing snapshot version,
+7. bytes representing mentioned block,
+8. bytes representing trie of the state.
 
 The node that makes a snapshot can serve it over http and new nodes can use this to speed up the catch up. Serving the snapshots
 over http is beyond the scope of Wasp and should be done in addition. Wasp is only responsible for making snapshots in local
-(configurable by `snapshots.localPath` parameter) folder and obtaining them when needed from the same local folder or from
+(configurable by `snapshots.localPath` parameter) folder and obtaining them on start when needed from the same local folder or from
 configured (by `snapshots.networkPaths` parameter) http addresses. A folder, serving snapshot files over http must contain
 `INDEX` file with new line separated list of snapshot file names.
 
-To know, which states are available as snapshots, Wasp periodically (see `snapshots.updatePeriod` parameter) updates the known
-snapshot list: it scans local folder and all the network addresses. In local folder it reads all the files with names that satisfy
-search pattern `*-*.snap`. In each network location Wasp reads all the files listed in `INDEX` file of that location. Wasp reads
-a state index and a commitment from the contents of these files. File names are not used to obtain this information and full
-snapshot files are not (down)loaded on this update. The obtained state index and commitment together with exact address
-of the snapshot are stored in known snapshot list. When snapshot is needed by state manager, the list of known snapshots
-is checked and the attempt to (down)load the snapshot from one of the locations in the list is made. If it fails, another
-location is tried. If obtaining snapshot from all the locations fail, it is assumed that the snapshot is not available.
+If chain is started with empty database (usually, if the database hasn't yet been created or it was deleted), the node checks, if
+it can load a snapshot: it scans local folder and all the network addresses for available snapshot files. In local folder it reads
+all the files with names that satisfy search pattern `*-*.snap`. In each network location Wasp reads all the files listed in `INDEX`
+file of that location. Wasp reads a state index and a commitment from the contents of these files. File names are not used to obtain
+this information and full snapshot files are not (down)loaded yet. Among all the available snapshot files, the node chooses the one
+with largest state index and loads it to the store. If there are several files with the same largest state index, the node tries
+to load them one by one starting from the local ones until one snapshot is loaded correctly. If loading fails for all candidates,
+the node is started with empty database.
```

Comment:
```suggestion
If a chain starts with an empty database (usually if the database hasn't been created yet or was deleted), the node checks if
it can load a snapshot: it scans the local folder and all the network addresses for available snapshot files. In the local folder, it reads
all the files with names that satisfy the search pattern `*-*.snap`. In each network location, Wasp reads all the files listed in `INDEX`
file of that location. Wasp reads a state index and a commitment from the contents of these files. File names are not used to obtain
this information, and full snapshot files are not (down)loaded yet. The node chooses 
the one with the largest state index and loads it to the store among all available snapshot files. If several files have the same largest state index, 
the node loads them one by one, starting from the local ones until one snapshot is loaded correctly. If loading fails for all candidates,
the node will start with an empty database.
```